python/ray/serve/router.py

```diff
@@ -161,8 +167,24 @@ def _all_query_refs(self):
     def _drain_completed_object_refs(self) -> int:
         refs = self._all_query_refs
         done, _ = ray.wait(refs, num_returns=len(refs), timeout=0)
-        for replica_in_flight_queries in self.in_flight_queries.values():
-            replica_in_flight_queries.difference_update(done)
+        replicas_to_remove = []
+        for replica_info, replica_in_flight_queries in self.in_flight_queries.items():
+            completed_queries = replica_in_flight_queries.intersection(done)
+            if len(completed_queries):
+                try:
+                    ray.get(completed_queries)
+                except RayActorError:
+                    replicas_to_remove.append(replica_info)
+                except Exception:
+                    pass  # ??? we never handled this before
```

Comment:
i think we should just ignore the exception here to keep in line with existing behavior; additionally the error should be propagated the user facing object_ref

Reply:
i'm afraid about crazy spammy log here :(

Reply:
What if we pass on `RayTaskError` and log any other `Exception`? That should really only be unknown system-level errors.